src/components/run-button-container.js

```diff
@@ -221,292 +192,287 @@ export function RunButtonContainer({ studyUuid, currentNode, disabled }) {
-                )
-            )
+        startComputationAsync(
+            ComputingType.SECURITY_ANALYSIS,
+            () => {
+                // close the contingency list selection window
+                setShowContingencyListSelector(false);
+                setComputationStopped(false);
+            },
+            () =>
+                startSecurityAnalysis(
+                    studyUuid,
+                    currentNode?.id,
+                    contingencyListNames
+                ),
+            () => {},
+            null,
+            null
         );
     };
 
     const handleStartDynamicSimulation = (dynamicSimulationConfiguration) => {
-        // close the dialog
-        setShowDynamicSimulationParametersSelector(false);
-
-        setComputationStopped(false);
-        dispatch(
-            setComputingStatus(
-                ComputingType.DYNAMIC_SIMULATION,
-                RunningStatus.RUNNING
-            )
+        startComputationAsync(
+            ComputingType.DYNAMIC_SIMULATION,
+            () => {
+                // close the dialog
+                setShowDynamicSimulationParametersSelector(false);
+                setComputationStopped(false);
+            },
+            () =>
+                startDynamicSimulation(
+                    studyUuid,
+                    currentNode?.id,
+                    dynamicSimulationConfiguration
+                ),
+            () => {},
+            null,
+            'DynamicSimulationRunError'
         );
-        // start server side dynamic simulation
-        return startDynamicSimulation(
-            studyUuid,
-            currentNode?.id,
-            dynamicSimulationConfiguration
-        ).catch((error) => {
-            dispatch(
-                setComputingStatus(
-                    ComputingType.DYNAMIC_SIMULATION,
-                    RunningStatus.FAILED
-                )
-            );
-            snackError({
-                messageTxt: error.message,
-                headerId: 'DynamicSimulationRunError',
-            });
-        });
     };
 
-    const startComputation = (action) => {
-        if (action === runnable[ComputingType.LOADFLOW]) {
-            dispatch(
-                setComputingStatus(
-                    ComputingType.LOADFLOW,
-                    RunningStatus.RUNNING
-                )
-            );
-            setComputationStopped(false);
-            startLoadFlow(
-                studyUuid,
-                currentNode?.id,
-                limitReductionParam / 100.0
-            )
-                .then(setRanLoadflow(true))
-                .catch((error) => {
-                    dispatch(
-                        setComputingStatus(
-                            ComputingType.LOADFLOW,
-                            RunningStatus.FAILED
-                        )
+    const runnables = useMemo(() => {
+        function actionOnRunnables(type, fnStop) {
+            fnStop.finally(() => {
```

Comment:
```suggestion
            fnStop().finally(() => {
```
There's also an error here when I try to interrupt a computation

Reply:
It seem I've a problem...